### ISSUE TITLE
Getting mean loss = nan, perplexity = inf while training 

### ISSUE THREAD

**Issue by shuheif** [bshao001/ChatLearner#59]
Hi @bshao001, thank you so much for this wonderful repository! I'm a beginner in machine learning and am learning a lot from your work! 

While training the model, I'm getting mean loss = nan and perplexity = inf, and the model never learns. I have tried to change batch_size to 32 and removed reddit dataset but didn't solve the issue. I run tokenizeddata.py with uncommenting the debugging code, and I think it worked just fine. Also, I tried the pre-trained model, and that works without any issues when I run botui.py. This issue happens on my PC with GPU (Tesla P100, which I believe has a good compute capability) but not on my laptop. I'm guessing my environment is the reason, but I have no idea where to check/fix. I would really appreciate if anyone could give me any kinds of suggestions or comments to solve this issue. Thanks in advance!

GPU: NVIDIA Corporation GP100GL [Tesla P100 PCIe 16GB]
NVRM version: NVIDIA UNIX x86_64 Kernel Module  396.51  Tue Jul 31 10:43:06 PDT 2018
GCC version:  gcc version 5.4.0 20160609 (Ubuntu 5.4.0-6ubuntu1~16.04.10)
Cuda compiler: 9.2
OS: Ubuntu 16.04.05 LTS 4.15.0-30-lowlatency
Python 3.5.2
TensorFlow 1.8.0
 
Below is the part of the log that I got. Though I got bunch of "CPU:0" in the log, my GPU seems to be working as well.
***
dynamic_seq2seq/encoder/rnn/while/rnn/multi_rnn_cell/cell_0/gru_cell/Const: (Const): /job:localhost/replica:0/task:0/device:GPU:0
dynamic_seq2seq/encoder/rnn/while/rnn/multi_rnn_cell/cell_0/gru_cell/concat/axis: (Const): /job:localhost/replica:0/task:0/device:GPU:0
dynamic_seq2seq/encoder/rnn/while/rnn/multi_rnn_cell/cell_0/dropout/random_uniform/max: (Const): /job:localhost/replica:0/task:0/device:GPU:0
dynamic_seq2seq/encoder/rnn/while/rnn/multi_rnn_cell/cell_0/dropout/random_uniform/min: (Const): /job:localhost/replica:0/task:0/device:GPU:0
dynamic_seq2seq/encoder/rnn/while/rnn/multi_rnn_cell/cell_0/dropout/keep_prob: (Const): /job:localhost/replica:0/task:0/device:GPU:0
dynamic_seq2seq/encoder/rnn/while/add/y: (Const): /job:localhost/replica:0/task:0/device:GPU:0
Adam/value: (Const): /job:localhost/replica:0/task:0/device:CPU:0
buffer_size_18: (Const): /job:localhost/replica:0/task:0/device:CPU:0
num_parallel_calls_1: (Const): /job:localhost/replica:0/task:0/device:CPU:0
buffer_size_17: (Const): /job:localhost/replica:0/task:0/device:CPU:0
num_parallel_calls: (Const): /job:localhost/replica:0/task:0/device:CPU:0
seed2: (Const): /job:localhost/replica:0/task:0/device:CPU:0
seed: (Const): /job:localhost/replica:0/task:0/device:CPU:0
buffer_size_16: (Const): /job:localhost/replica:0/task:0/device:CPU:0
buffer_size_15: (Const): /job:localhost/replica:0/task:0/device:CPU:0
buffer_size_14: (Const): /job:localhost/replica:0/task:0/device:CPU:0
buffer_size_13: (Const): /job:localhost/replica:0/task:0/device:CPU:0
buffer_size_12: (Const): /job:localhost/replica:0/task:0/device:CPU:0
buffer_size_11: (Const): /job:localhost/replica:0/task:0/device:CPU:0
buffer_size_10: (Const): /job:localhost/replica:0/task:0/device:CPU:0
buffer_size_9: (Const): /job:localhost/replica:0/task:0/device:CPU:0
buffer_size_8: (Const): /job:localhost/replica:0/task:0/device:CPU:0
buffer_size_7: (Const): /job:localhost/replica:0/task:0/device:CPU:0
buffer_size_6: (Const): /job:localhost/replica:0/task:0/device:CPU:0
compression_type_2: (Const): /job:localhost/replica:0/task:0/device:CPU:0
filenames_2: (Const): /job:localhost/replica:0/task:0/device:CPU:0
count_1: (Const): /job:localhost/replica:0/task:0/device:CPU:0
buffer_size_5: (Const): /job:localhost/replica:0/task:0/device:CPU:0
buffer_size_4: (Const): /job:localhost/replica:0/task:0/device:CPU:0
buffer_size_3: (Const): /job:localhost/replica:0/task:0/device:CPU:0
compression_type_1: (Const): /job:localhost/replica:0/task:0/device:CPU:0
filenames_1: (Const): /job:localhost/replica:0/task:0/device:CPU:0
count: (Const): /job:localhost/replica:0/task:0/device:CPU:0
buffer_size_2: (Const): /job:localhost/replica:0/task:0/device:CPU:0
buffer_size_1: (Const): /job:localhost/replica:0/task:0/device:CPU:0
buffer_size: (Const): /job:localhost/replica:0/task:0/device:CPU:0
compression_type: (Const): /job:localhost/replica:0/task:0/device:CPU:0
filenames: (Const): /job:localhost/replica:0/task:0/device:CPU:0
hash_table/Const: (Const): /job:localhost/replica:0/task:0/device:CPU:0
Const_1: (Const): /job:localhost/replica:0/task:0/device:CPU:0
Const: (Const): /job:localhost/replica:0/task:0/device:CPU:0
string_to_index/hash_table/table_init/asset_filepath: (Const): /job:localhost/replica:0/task:0/device:CPU:0
string_to_index/hash_table/Const: (Const): /job:localhost/replica:0/task:0/device:CPU:0
\# Training loop started @ 2018-08-15 15:56:50
\# Finished epoch  1 @ step  1396 @ 2018-08-15 16:00:51. In the epoch, learning rate = 0.000800, mean loss = nan, perplexity =      inf, and 240.48 seconds elapsed.

**Comment by bshao001** [comment]
@shuheif I am not sure what can cause this weird problem either. Other than re-install your whole environment, one thing you can try is to insert tf.Print statements inside the model, especially the part of the loss function. This might help you figure out the exact place of the problem happening. Good luck!

**Comment by shuheif** [comment]
@bshao001 Thank you so much for your comment and I'm sorry for my late reply. I'm looking for the exact place of the issue using print statement. I will post here when I find it out.
